app/models/squix/print_job_wrapper.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Squix
+  # Squix::PrintJobWrapper
+  class PrintJobWrapper
+    include ActiveModel::Model
+
+    attr_accessor :printer_name, :label_template_name, :label_template_id, :labels, :copies, :errors
+
+    # validate
+
+    def initialize(printer_name, label_template_name, label_template_id, labels, copies)
+      @printer_name = printer_name
+      @label_template_name = label_template_name
+      @label_template_id = label_template_id
+      @labels = labels
+      @copies = copies.to_i
+      @errors = []
+    end
+
+    def print
+      # return false unless valid?
+
+      case printer_type
+      when 'Toshiba'
+        print_to_toshiba
+      when 'Squix'
+        print_to_squix
+      else
+        errors << "Printer type #{printer_type} for printer #{printer_name} not recognised."
+        false
+      end
+    end
+
+    def printer_type
+      Printer.find_by(name: printer_name).type
+    end
+
+    def print_to_toshiba
+      unless label_template_id
+        @label_template_id = LabelTemplate.find_by(name: label_template_name).id
+      end
+
+      # labels here from LabWhere already includes the correct number of copies
+      body = { printer_name: printer_name, label_template_id: label_template_id, labels: labels }
+
+      print_job = LabelPrinter::PrintJob.build(body)
```

Comment:
You could make print job an attribute and then you could just check whether it is valid before executing which means you would not have to decide what type it is.